### ISSUE TITLE
Content nodes going "offline" during index.

### ISSUE THREAD

**Issue by jcsantosbr** [vespa-engine/vespa#21822]
**Describe the bug**

During indexing, some nodes in the cluster went "offline" with the below error in the logs:

```
1648203162.774891   <content_host> 7413    searchnode  stderr  warning vespa-proton-bin: /builddir/build/BUILD/vespa-7.557.46/searchlib/src/vespa/searchlib/attribute/postinglistattribute.cpp:63: void search::PostingListAttributeBase<P>::handle_load_posting_lists_and_update_enum_store(search::enumstore::EnumeratedPostingsLoader&) [with P = vespalib::btree::BTreeKeyData<unsigned int, int>]: Assertion `preve < elem.getEnum()' failed.
1648203178.712461   <content_host> 404/35  metricsproxy-container  Container.ai.vespa.metricsproxy.service.HttpMetricFetcher   warning Unable to get response from service 'searchnode:7413:RUNNING:entity/search/cluster.entity/21': Connect to http://localhost:19107/ [localhost/127.0.0.1, localhost/0:0:0:0:0:0:0:1] failed: Connection refused
1648203181.608671   <content_host> 383/41817   config-sentinel sentinel.sentinel.service   event   crash/1 name="searchnode" pid=7413 signal="Aborted"
1648203181.608735   <content_host> 383/41817   config-sentinel sentinel.sentinel.service   info    searchnode: incremented restart penalty to 1800.000 seconds
1648203181.608750   <content_host> 383/41817   config-sentinel sentinel.sentinel.service   info    searchnode: incremented restart penalty to 1800.000 seconds
1648203181.608767   <content_host> 383/41817   config-sentinel sentinel.sentinel.service   info    searchnode: will delay start by 1116.200 seconds
```

I’ve setup a cluster in a grouped distribution with 8 groups, 2 nodes per group (96 cores, 360GB ram).
I'm indexing 6 different schemas with about 50M, 10M, 10M, 2M, 1M and 200k documents.
Each document is basically composed of a similar structure to:

```
    document entityName {
        field entityId type string {
            indexing: summary | attribute
            rank: filter
        }

        field entityName type string {
            indexing: summary | attribute
        }

        field global_rank type float {
            indexing: summary | attribute
            attribute: fast-search
            rank: filter
        }

        field regional_rank type tensor<float>(country{}) {
            indexing: summary | attribute
            attribute: fast-search
            rank: filter
        }

        field availability_always_blocked type array<string> {
            indexing: summary
        }

        field availability_always_allowed type array<string> {
            indexing: summary | attribute
            attribute: fast-search
            rank: filter
        }

        struct AvailabilityRange {
            field key type string {}
            field start type long {}
            field end type long {}
        }

        field availability_scheduled type array<AvailabilityRange> {
            indexing: summary

            struct-field key {
                indexing: attribute
                attribute: fast-search
            }
            struct-field start {
                indexing: attribute
                attribute: fast-search
            }
            struct-field end {
                indexing: attribute
                attribute: fast-search
            }
        }

        field bert_embedding type tensor<float>(x[768]) {
            indexing: summary | attribute | index
            attribute {
                distance-metric: angular
            }
            index {
                hnsw {
                    max-links-per-node: 24
                    neighbors-to-explore-at-insert: 500
                }
            }
        }
    }
```

The cluster was not serving any traffic during this issue.

**To Reproduce**
1. Setup a cluster with the above setup
2. Start the indexing process.
3. After an unspecified time, some nodes in the cluster went down.

**Expected behavior**
All nodes to stay healthy after index

**Screenshots**

**Environment (please complete the following information):**
 - OS: Linux
 - Infrastructure: GCE in GCP.

**Vespa version**
Vespa 7.557.46

**Additional context**

We have this cluster in 3 different regions (Asia, Europe, America): At least one node in each region was affected.

**Comment by geirst** [comment]
In parallel we will work on improving the radix sort code to handle more than 4G values.

**Comment by jcsantosbr** [comment]
Hi @geirst,
I'm sending you the response you asked from our biggest schema, most likely to be the culprit.

I see that the attribute availability_always_allowed has 4963071483 total value count which seems to confirm your hypothesis.

Thanks for the quick response!

[entity_atributes.txt](https://github.com/vespa-engine/vespa/files/8351951/entity_atributes.txt)

**Comment by geirst** [comment]
Thanks @jcsantosbr!
Yes, this confirms our hypothesis. The _totalValueCount_ above 4G is the problem. Will keep you posted when we have a new Vespa version with a fix. ETA is sometime next week. Until then the affected nodes will unfortunately not be able to start.

**Comment by jcsantosbr** [comment]
Great to confirm you found the issue.
We will try to put a workaround in place until the fix is ready.
Thanks again!

**Comment by baldersheim** [comment]
The issue was solved in #21832 and is part of 7.565.45 and later builds.
The next rpm build is due on Monday 2022-03-28 if all build and test pipelines converge.

**Comment by baldersheim** [comment]
Open source release 7.565.113 is out and contains the above fix.
